src/components/InputField/InputField.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div
+    class="input-field group flex flex-row items-center gap-4 hover:bg-dark-charcoal-06 h-12 p-0.5px focus-within:p-0 border focus-within:border-1.5 border-dark-charcoal-20 rounded-sm overflow-hidden focus-within:border-pink"
+    :class="[
+      {
+        // Padding is set to 1.5px to accommodate the border that will appear later.
+        'border-s-0 ps-1.5px rounded-s-none': connectionSides.includes('start'),
+        'border-e-0 pe-1.5px rounded-e-none': connectionSides.includes('end'),
+      },
+    ]"
+  >
+    <!-- eslint-disable vuejs-accessibility/form-control-has-label -->
+    <!-- The `inputId` prop is provided so that the user of the component can associate a label. -->
+    <input
+      v-bind="$attrs"
+      :id="inputId"
+      v-model="text"
+      :type="type"
+      class="flex-grow leading-none font-semibold bg-tx ms-4 h-full focus:outline-none"
+    />
+    <!-- eslint-enable vuejs-accessibility/form-control-has-label -->
+    <div
+      class="info font-semibold text-xs text-dark-charcoal-70 group-hover:text-dark-charcoal me-4"
+    >
+      <!-- @slot Extra information goes here -->
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script>
+import { computed } from '@nuxtjs/composition-api'
+
+/**
+ * Provides a control to enter text as input.
+ */
+export default {
+  name: 'InputField',
+  inheritAttrs: false,
+  model: {
+    prop: 'value',
+    event: 'input',
+  },
+  props: {
+    /**
+     * the textual content of the input field
+     */
+    value: {
+      type: String,
+      default: '',
+    },
+    /**
+     * the ID to associate with the internal `<input>` element; This should be
+     * used to associate a label and is recommended for a11y.
+     */
+    inputId: {
+      type: String,
+    },
```

Comment:
Given that a label is required for accessibility, most input field components will combine the label and input together to ensure that the label is always rendered. Gutenberg's Input component does this and has a `visuallyHideLabel` prop that makes it only visible for screen readers. If we do that on the component level then there's no way for us to create an inaccessible input.

Reply:
I made this change in b34a843. Boiling down these changes:
- `inputId` is gone (you can directly attach `id` via `$attrs`)
- new prop `labelText` allows setting an SR-only label inside the `<label>` that now wraps `<input>`
- skipping this prop means the `<span>` is not rendered (allowing you to set an `id` and BYO label)

I went with the SR-only label instead of a visual one because we don't have any labelled fields in the design library.